### ISSUE TITLE
[Bug]: alter table fail because of syntax error of backgroud SQL

### ISSUE THREAD

**Issue by jensenojs** [matrixorigin/matrixone#15434]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

123d3af09

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```


### Actual Behavior

```sql

mysql> alter table index03 add unique key `tempKey`(col1,col2) comment 'abcTest';

ERROR 1105 (HY000): You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at line 1 column 35 near " alter.index03 group by serial(col1,col2) having count(*) > 1 and serial(col1,col2) is not null;";

```

<img width="1511" alt="image" src="https://github.com/matrixorigin/matrixone/assets/56761542/d328a675-51a5-4f64-b3c7-7327e2bbde74">


### Expected Behavior

_No response_

### Steps to Reproduce

```Markdown
mysql> create database `alter`;                                                                                                                                                                                                                   
Query OK, 1 row affected (0.05 sec)

mysql> use `alter`;
Database changed
mysql> create table index03(col1 int, col2 int, col3 int);
Query OK, 0 rows affected (0.09 sec)

mysql> insert into index03 values (1,2,1);                                                                                                                                                                                                               Continuum status: 
Query OK, 1 row affected (0.03 sec)

mysql> insert into index03 values (1,2,1);
Query OK, 1 row affected (0.01 sec)

mysql> alter table index03 add unique key `tempKey`(col1,col2) comment 'abcTest';
ERROR 1105 (HY000): You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at line 1 column 35 near " alter.index03 group by serial(col1,col2) having count(*) > 1 and serial(col1,col2) is not null;";
```


### Additional information

_No response_

**Comment by Ariznawlll** [comment]
以任意保留关键字建一个db，然后use这个db，再进行alter操作，都有这个问题：

create database `constraint`;
use `constraint`;
create table index03(col1 int, col2 int, col3 int);
insert into index03 values (1,2,1);
insert into index03 values (1,2,1);
alter table index03 add unique key `tempKey`(col1,col2) comment 'abcTest';

<img width="1812" alt="image" src="https://github.com/matrixorigin/matrixone/assets/108530700/b99fbde1-bf1e-4866-9391-18f4b0cd7b87">

**Comment by jensenojs** [comment]
<img width="374" alt="企业微信截图_0fe44d4e-e32d-4a7b-b9e4-2bc3da5db38a" src="https://github.com/matrixorigin/matrixone/assets/56761542/fe47b0fb-89fe-41c9-b8ea-20fbffae0949">

@Ariznawlll 这个情况也值得关注, 甚至都不好说是不是一个issue反映的问题

**Comment by jensenojs** [comment]
pr已经合并, 相关测试用例魏璐已经补充.

**Comment by aressu1985** [comment]
fixed:275af9cc9a4d113bd5ddcb5381eb9a9d15e0dd87

mysql> create database `alter`; 
Query OK, 1 row affected (0.05 sec)

mysql> use `alter`;
Database changed
mysql> create table index03(col1 int, col2 int, col3 int);
Query OK, 0 rows affected (0.05 sec)

mysql> insert into index03 values (1,2,1);   
Query OK, 1 row affected (0.03 sec)

mysql> insert into index03 values (1,2,1);
Query OK, 1 row affected (0.06 sec)

mysql> alter table index03 add unique key `tempKey`(col1,col2) comment 'abcTest';
ERROR 1062 (HY000): Duplicate entry '(1,2)' for key '__mo_index_idx_col'
